test/Mono.Linker.Tests.Cases/DataFlow/ArrayDataFlow.cs

```diff
@@ -205,6 +207,246 @@ static void TestArrayResetAfterAssignment ()
 
 		static Type[] _externalArray;
 
+		static class TestMultiDimensionalArray
+		{
+			public static void Test ()
+			{
+				TestArrayWithInitializerOneElementStaticType ();
+				TestArrayWithInitializerOneElementParameter (typeof (TestType));
+				TestArrayWithInitializerMultipleElementsStaticType ();
+				TestArrayWithInitializerMultipleElementsMix<TestType> (typeof (TestType));
+
+				TestArraySetElementOneElementStaticType ();
+				TestArraySetElementOneElementParameter (typeof (TestType));
+				TestArraySetElementMultipleElementsStaticType ();
+				TestArraySetElementMultipleElementsMix<TestType> (typeof (TestType));
+
+				TestArraySetElementAndInitializerMultipleElementsMix<TestType> (typeof (TestType));
+
+				TestGetElementAtUnknownIndex ();
+
+				// Array reset - certain operations on array are not tracked fully (or impossible due to unknown inputs)
+				// and sometimes the only valid thing to do is to reset the array to all unknowns as it's impossible
+				// to determine what the operation did to the array. So after the reset, everything in the array
+				// should be treated as unknown value.
+				TestArrayResetStoreUnknownIndex ();
+				TestArrayResetGetElementOnByRefArray ();
+				TestArrayResetAfterCall ();
+				TestArrayResetAfterAssignment ();
+			}
+
+			[ExpectedWarning ("IL2062", nameof (DataFlowTypeExtensions.RequiresPublicMethods), ProducedBy = ProducedBy.Trimmer)]
+			// Bug
+			[ExpectedWarning ("IL2062", nameof (DataFlowTypeExtensions.RequiresAll), ProducedBy = ProducedBy.Trimmer)]
```

Comment:
Why bug? I think the intent currently is to simply not track multi dimensional arrays at all...

Reply:
I would not say never... but it's definitely VERY low on the list... I mean all we care about are `Type` values really - why would you put one into a multi dimensional array (I know, famous last words).
Not counting the fact that I haven't seen multi dimensional array in customer code for a long time...